src/extension/token/voting/VotingERC721.sol

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: Apache 2.0
+pragma solidity ^0.8.0;
+
+import {Historical} from "./Historical.sol";
+import {ModularExtension} from "../../../ModularExtension.sol";
+import {IERC721A, ERC721A, ERC721AQueryable} from "@erc721a/extensions/ERC721AQueryable.sol";
+import {SafeCastLib} from "@solady/utils/SafeCastLib.sol";
+import {EIP712} from "@solady/utils/EIP712.sol";
+import {ECDSA} from "@solady/utils/ECDSA.sol";
+
+library VotingStorage {
+    /// @custom:storage-location erc7201:token.voting
+    bytes32 public constant VOTING_STORAGE_POSITION =
+        keccak256(abi.encode(uint256(keccak256("token.voting")) - 1)) & ~bytes32(uint256(0xff));
+
+    struct Data {
+        mapping(address account => address) delegatedTo;
+        mapping(address delegatee => Historical.Timeline) delegatedVotes;
+        mapping(address account => uint256) nonces;
+        Historical.Timeline totalVotes;
+    }
+
+    function data() internal pure returns (Data storage data_) {
+        bytes32 position = VOTING_STORAGE_POSITION;
+        assembly {
+            data_.slot := position
+        }
+    }
+}
+
+contract VotingERC721 is ERC721AQueryable, EIP712, ModularExtension {
```

Comment:
If Voting is an extension it should not inherits ERC721AQueryable, because it should be installed on a ERC721 core contract

Reply:
Think of Core = ERC20, ERC721, ERC721A, ERC1155
Extensions = additional capabilities and functionalities to install on the Core contracts.

Extensions can be specific to a token core spec, or agnostic supporting all token ercs

Reply:
Reason it is inheriting ERC721A is to gain access to `balanceOf` function from ERC721. It's needed in the following function in order to calculate the number of votes that can be delegated from an account

```solidity

    /// @notice Returns the balance of `account`.
    function _getVotingUnits(address account) internal view returns (uint256) {
        return balanceOf(account);
    }
```

Reply:
most recent commit should now resolve this